runtimes/eoapi/stac/eoapi/stac/client.py

```diff
@@ -367,43 +437,115 @@ async def item_collection(
+
+        base_args = {
+            "collections": [collection_id],
+            "bbox": bbox,
+            "datetime": datetime,
+            "limit": limit,
+            "token": token,
+            "query": orjson.loads(unquote_plus(query)) if query else query,
+        }
+        clean = self._clean_search_args(
+            base_args=base_args,
+            filter_query=filter_expr,
+            filter_lang=filter_lang,
+            fields=fields,
+            sortby=sortby,
+        )
+
+        search_request = self.pgstac_search_model(**clean)
+        item_collection = await self._search_base(search_request, request=request)
+        item_collection["links"] = await ItemCollectionLinks(
+            collection_id=collection_id, request=request
+        ).get_links(extra_links=item_collection["links"])
+
+        # Additional Headers for StreamingResponse
+        additional_headers = {}
+        links = item_collection.get("links", [])
+        next_link = next(filter(lambda link: link["rel"] == "next", links), None)
+        prev_link = next(
+            filter(lambda link: link["rel"] in ["prev", "previous"], links), None
+        )
+        if next_link or prev_link:
+            additional_headers["Link"] = ",".join(
+                [
+                    f'<{link["href"]}>; rel="{link["rel"]}"'
+                    for link in [next_link, prev_link]
+                    if link
+                ]
+            )
+
         if output_type == MimeTypes.html:
-            items["id"] = collection_id
+            item_collection["id"] = collection_id
             return create_html_response(
                 request,
-                items,
+                item_collection,
                 template_name="items",
                 title=f"{collection_id} items",
             )
 
-        return items
+        elif output_type == MimeTypes.csv:
+            return StreamingResponse(
+                items_to_csv_rows(item_collection["features"]),
+                media_type=MimeTypes.csv,
+                headers={
+                    "Content-Disposition": "attachment;filename=items.csv",
+                    **additional_headers,
+                },
+            )
+
+        elif output_type == MimeTypes.geojsonseq:
+            return StreamingResponse(
+                (orjson.dumps(f) + b"\n" for f in item_collection["features"]),
+                media_type=MimeTypes.geojsonseq,
+                headers={
+                    "Content-Disposition": "attachment;filename=items.geojson",
+                    **additional_headers,
+                },
+            )
+
+        # If we have the `fields` extension enabled
+        # we need to avoid Pydantic validation because the
+        # Items might not be a valid STAC Item objects
+        if fields := getattr(search_request, "fields", None):
+            if fields.include or fields.exclude:
+                return JSONResponse(item_collection)  # type: ignore
```

Comment:
We HAVE to re-write the whole method in our custom runtime because if a user passed `fields=` then the orignal method was returning a JSONResponse, which would then need to be parsed back if we wanted to return an HTML response